t2v_metrics/models/vqascore_models/gpt4v_model.py

```diff
@@ -44,97 +59,80 @@ def __init__(self,
+                ]
+
             completion = self.client.chat.completions.create(
                 model=self.model_name,
-                messages= [
-                {
-                    "role": "user",
-                    "content": [
-                    {
-                        "type": "text",
-                        "text": question
-                    },
-                    {
-                        "type": "image_url",
-                        "image_url": {
-                        "url": f"data:image/{image['type']};base64,{image['base64']}"
-                        }
-                    }
-                    ]
-                }
-                ],
+                messages=[{"role": "user", "content": content}],
                 logprobs=True,
                 top_logprobs=self.top_logprobs,
-                # logit_bias={yes_token:50, no_token:50}
             )
-        except:
-            print(f"Warning: completion not generated for image: {image['path']} and question: {question} and answer: {answer}")
-            print(f"Trying again with the same image")
-            try:
-                completion = self.client.chat.completions.create(model=self.model_name, messages= [{"role": "user","content": [{"type": "text", "text": question},{ "type": "image_url","image_url": {"url": f"data:image/{image['type']};base64,{image['base64']}"}}]}],logprobs=True,top_logprobs=self.top_logprobs,)
-            except:
-                print(f"Failed image: {image['path']} and question: {question} and answer: {answer}")
-                return torch.Tensor([0.0])
-
-        # print(completion.choices[0].message)
-        # print(completion.choices[0].logprobs)
-        # print(completion.choices[0].logprobs.content[0])
-        is_generated = False
+        except Exception as e:
+            print(f"Warning: completion not generated for {data['path']} and question: {question}")
+            print(f"Error: {str(e)}")
+            return torch.Tensor([0.0])
+
         for top_logprob in completion.choices[0].logprobs.content[0].top_logprobs:
+            print(answer)
             if top_logprob.token == answer:
-                is_generated = True
                 return torch.Tensor([top_logprob.logprob]).exp()
-        if not is_generated:
-            print(f"Warning: answer not generated for image: {image['path']} and question: {question} and answer: {answer}")
-            print(completion.choices[0].logprobs.content[0].top_logprobs)
-            return torch.Tensor([0.0])
+        
+        print(f"Warning: answer not generated for {data['path']} and question: {question}")
+        print(completion.choices[0].logprobs.content[0].top_logprobs)
+        return torch.Tensor([0.0])
 
-    # @torch.no_grad()
-    # @torch.autocast(device_type='cuda', dtype=torch.bfloat16)
     def forward(self,
-                images: List[str],
+                paths: List[str],
```

Comment:
Will this change be compatible with other higher-level modules?

Reply:
I guess we can stick with your new module -- if users want to pass in images then they use `images`, but if they want to pass in videos they can use `videos`. We can add a few assert statements to make sure the user didn't pass in both `images` and `videos`

Reply:
Yes, this should work fine since whether it is `images` or `videos` passed by the user, the preprocessing is done at the model level.